pkg/clientconfighelpers/helpers.go

```diff
@@ -8,50 +8,87 @@ package clientconfighelpers
 import (
 	"encoding/base64"
 	"os"
+	"runtime"
+	"strconv"
 
-	ctlimg "github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
-
+	regname "github.com/google/go-containerregistry/pkg/name"
 	"github.com/pkg/errors"
 
 	"github.com/vmware-tanzu/tanzu-cli/pkg/configpaths"
 	"github.com/vmware-tanzu/tanzu-cli/pkg/constants"
+	configlib "github.com/vmware-tanzu/tanzu-plugin-runtime/config"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
 )
 
-// GetCustomRepositoryCaCertificateForClient returns CA certificate to use with cli client
-// This function reads the CA certificate from following variables in decreasing order of precedence:
-// 1. PROXY_CA_CERT
-// 2. TKG_PROXY_CA_CERT
-// 3. TKG_CUSTOM_IMAGE_REPOSITORY_CA_CERTIFICATE
-func GetCustomRepositoryCaCertificateForClient() ([]byte, error) {
-	caCert := ""
-	var errProxyCACert, errTkgProxyCACertValue, errCustomImageRepoCACert error
-	var proxyCACertValue, tkgProxyCACertValue, customImageRepoCACert string
+type RegistryCertOptions struct {
```

Comment:
Can all the updated things in this file be moved under the `pkg/registry` package? Using `pkg/clientconfighelpers` package for this stuff feels weird.

Reply:
updated.